include/envoy/http/filter.h

```diff
@@ -348,6 +348,16 @@ class StreamDecoderFilterCallbacks : public virtual StreamFilterCallbacks {
   // Note that HttpConnectionManager sanitization will *not* be performed on the
   // recreated stream, as it is assumed that sanitization has already been done.
   virtual bool recreateStream() PURE;
+
+  /**
+   * Called at the end of the stream, when all data has been encoded.
+   */
+  virtual void onEncodeComplete() PURE;
```

Comment:
I don't think this is exactly what you want. This class are callbacks for filters to call into the HCM. I think you actually want to call a function in the filter. For this we have 2 options:
1) You can define it directly on the filter class, and although virtual, have a default implementation that does nothing.
2) You could have filters register for the callback, like we do for watermark callbacks (for example).

I have a vague preference for (1) because these callbacks are pretty straightforward but I'm fine either way.

Reply:
I think you're referring to the `StreamDecoderFilter` class? I was wondering if that was a better home for these, so (1) seems fine with me.